### PR TITLE
TIP-221: domain layer extraction

### PR DESCRIPTION
**Pull request by jjanvier**
This PR moves a bunch of business stuff into the catalog component.

before: 27 coupling rules broken
after: 27 coupling rules broken (the same)

What's moved here:
* product events
* exceptions
* completeness generator interface
* PQB :)

For the record, all things related to the command `pim:product:query-help` have been moved to `CatalogBundle\Command\ProductQueryHelp`

```
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Changelog updated                 | Y
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -
```

### CODE REVIEW

**Comment by juliensnz** [comment]
GTM

File: CHANGELOG-1.6.md [hunk 1]
```diff
@@ -6,6 +6,13 @@
 
 ## BC breaks
 
+- Move `Pim\Bundle\CatalogBundle\Query\Filter\DumperInterface` to `Pim\Bundle\CatalogBundle\Command\DumperInterface`
```
Comment (skeleton) [inline CHANGELOG-1.6.md]: add sed commands

File: src/Pim/Bundle/CatalogBundle/Builder/ProductBuilder.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace Pim\Bundle\CatalogBundle\Builder;
 
-use Pim\Bundle\CatalogBundle\Event\ProductEvents;
+use Pim\Component\Catalog\ProductEvents;
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Builder/ProductBuilder.php]: order ^^

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Condition/CriteriaCondition.php [hunk 3]
```diff
@@ -93,7 +93,7 @@ protected function prepareMultiCriteriaCondition($field, $operator, $value)
      * @param string       $operator the operator used to filter
      * @param string|array $value    the value(s) to filter
      *
-     * @throws \Pim\Bundle\CatalogBundle\Exception\ProductQueryException
+     * @throws \Pim\Component\Catalog\Exception\ProductQueryException
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Condition/CriteriaCondition.php]: import namespace

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/AbstractFilter.php [hunk 4]
```diff
@@ -70,7 +70,7 @@ protected function getUniqueAlias($alias)
      * @param string|array $operator the operator used to filter
      * @param string|array $value    the value(s) to filter
      *
-     * @throws \Pim\Bundle\CatalogBundle\Exception\ProductQueryException
+     * @throws \Pim\Component\Catalog\Exception\ProductQueryException
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/AbstractFilter.php]: import it

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/AbstractFilter.php [hunk 5]
```diff
@@ -89,7 +89,7 @@ protected function prepareCriteriaCondition($field, $operator, $value)
      * @param string             $locale    the locale
      * @param string             $scope     the scope
      *
-     * @throws \Pim\Bundle\CatalogBundle\Exception\ProductQueryException
+     * @throws \Pim\Component\Catalog\Exception\ProductQueryException
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/AbstractFilter.php]: same

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Sorter/BaseSorter.php [hunk 6]
```diff
@@ -141,7 +141,7 @@ public function addFieldSorter($field, $direction, $locale = null, $scope = null
      * @param string             $locale    the locale
      * @param string             $scope     the scope
      *
-     * @throws \Pim\Bundle\CatalogBundle\Exception\ProductQueryException
+     * @throws \Pim\Component\Catalog\Exception\ProductQueryException
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Sorter/BaseSorter.php]: same

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Sorter/EntitySorter.php [hunk 7]
```diff
@@ -91,7 +91,7 @@ public function addAttributeSorter(AttributeInterface $attribute, $direction, $l
      * @param string             $locale    the locale
      * @param string             $scope     the scope
      *
-     * @throws \Pim\Bundle\CatalogBundle\Exception\ProductQueryException
+     * @throws \Pim\Component\Catalog\Exception\ProductQueryException
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Sorter/EntitySorter.php]: same

**Comment by skeleton** [comment]
please, fix namespaces order with php-cs-fixer
gtm once fixed

**Comment by nidup** [comment]
Add sed cmds then GTM

File: CHANGELOG-1.6.md (hunk 1 above)
Comment (jjanvier) [inline CHANGELOG-1.6.md]: done

**Pull request merged**